sqlglot/dialects/bigquery.py

```diff
@@ -358,6 +358,17 @@ class Parser(parser.Parser):
 
         NULL_TOKENS = {TokenType.NULL, TokenType.UNKNOWN}
 
+        STATEMENT_PARSERS = {
+            **parser.Parser.STATEMENT_PARSERS,
+            TokenType.END: lambda self: self._parse_as_command(self._prev),
+            TokenType.FOR: lambda self: self._parse_for(),
+        }
+
+        def _parse_for(self) -> exp.ForIn:
+            this = self._parse_range()
+            self._match_text_seq("DO")
+            return self.expression(exp.ForIn, this=this, expression=self._parse_statement())
```

Comment:
can you do multiple statements separated by semi colon?

Reply:
we split statements into chunks on `;` at the top level `_parse` method so it's not trivial

Reply:
this should also answer:

> why do you have this? shouldn't this be a part of the for in parser?